File: backend/app/models/collections.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class Collection(BaseModel):
    id: int
    user_id: int
    name: str
    is_default: bool
    created_at: Optional[datetime] = None


class CollectionGame(BaseModel):
    id: int
    collection_id: int
    game_id: int
    added_at: Optional[datetime] = None


class CreateCollectionRequest(BaseModel):
    name: str


class RenameCollectionRequest(BaseModel):
    name: str


class AddGameToCollectionRequest(BaseModel):
    game_id: int

File: frontend/src/types/collections.ts
export type Collection = {
	id: number;
	user_id: number;
	name: string;
	is_default: boolean;
	created_at: string | null;
};

export type CollectionGame = {
	id: number;
	collection_id: number;
	game_id: number;
	added_at: string | null;
};

export type CreateCollectionRequest = {
	name: string;
};

export type RenameCollectionRequest = {
	name: string;
};

export type AddGameToCollectionRequest = {
	game_id: number;
};
